### DISCUSSION TITLE
I don't know where to change  Multi ControlNet: Max models amount (requires restart)

### DISCUSSION THREAD

**Discussion by hl123-123** [Mikubill/sd-webui-controlnet#471] [General]
I don't know the settings is where.

**Comment by ljleb** [comment]
Go to Settings tab > ControlNet sub tab > you'll find a slider under that name, move it around. Should add this in some form of documentation along with other options.

**Reply by hl123-123** [reply]
thank you a lot

**Comment by BEpresent** [comment]
Is there any way to change that setting via some API call, without starting the webui ? 

When I start i with `--api --nowebui` the controlnet settings don't seem to be applied automatically.

**Reply by vandanchopra** [reply]
+1 on this. I'd like to be able to change this setting through the API on a fresh install of a1111 in --nowebui mode.

**Reply by ljleb** [reply]
It appears you can use the route `POST /sdapi/v1/options`  with the payload:
```
{
  "control_net_max_models_num": N
}
```

Where `N` is the number of max controlnet units you want to instantiate. Ideally, this number shouldn't be needed for API calls, but I guess since the webui API integration does not allow to pass an arbitrary number of arguments to the extension, this is a valid workaround.

Note that for the setting to take effect, you need to restart the gradio interface. If you really want to control this setting in a fresh webui install before startup, consider setting `"control_net_max_models_num"` in config.json before startup.
